docs/development/standards.rst

```diff
@@ -35,70 +40,186 @@ Docker for builds
+
+``inv docker.up``
+    Starts all the containers needed to run Read the Docs completely.
+
+    * ``--no-search`` can be passed to disable search
+    * ``--init`` is used the first time this command is ran to run initial migrations, create an admin user,
+      setup Azurite containers, etc
+    * ``--no-reload`` makes all celery processes and django runserver
+      to use no reload and do not watch for files changes
+
+``inv docker.shell``
+    Opens a shell in a container (web by default).
+
+    * ``--running`` the shell is open in a container that it's already running
+    * ``--container`` specifies in which container the shell is open
+
+``inv docker.manage {command}``
+    Executes a Django management command in a container.
+
+    .. tip::
+
+       Useful when modifying models to run ``makemigrations``.
+
+``inv docker.down``
+    Stops and removes all containers running.
+
+    * ``--volumes`` will remove the volumes as well (database data will be lost)
+
+``inv docker.restart {containers}``
+    Restarts the containers specified (automatically restarts NGINX when needed).
+
+``inv docker.attach {container}``
+    Grab STDIN/STDOUT control of a running container.
+
+    .. tip::
+
+       Useful to debug with ``pdb``. Once the program has stopped in your pdb line,
+       you can run ``inv docker.attach web`` and jump into a pdb session
+       (it also works with ipdb and pdb++)
+
+``inv docker.test {arguments}``
+    Runs the test suite inside the container.
+
+``inv docker.pull``
+    Downloads and tags all the Docker images required for builders.
+
+
+Adding a new Python dependency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Docker image for the servers is built with the requirements defined in the ``master`` branch.
+In case you need to add a new Python dependency while developing,
+you can use the ``common/dockerfiles/entrypoints/common.sh`` script as shortcut.
+
+This script is run at startup on all the servers (web, celery, builder, proxito) which
+allows you to test your dependency without re-building the whole image.
+To do this, add the ``pip`` command required for your dependency in ``common.sh`` file:
+
+.. code-block:: bash
+
+   # common.sh
+   pip install my-dependency==1.2.3
+
+Once the PR that adds this dependency was merged into ``master``, you can rebuild the image
+so the dependency is added to the Docker image itself and it's not needed to be installed
+each time the container spins up.
+
+
+Adding support for ``-ext``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+   This is a temporary solution.
+
+
+.. warning::
+
+   ``readthedocs-ext`` is closed source for now and it's not available for non-core team members.
+
+Read the Docs extensions can be installed with the same pattern than `Adding a new Python dependency`_
+using the ``common.sh`` script. You can add the following line to the script:
+
+.. code-block:: bash
+
+   # common.sh
+   pip install -e ../readthedocs-ext/
```

Comment:
That could be a good addition if we can do it in the `Dockerfile`. I would try to avoid running this install every time the containers speed up.

I think we could check if `GITHUB_TOKEN` was passed in the `docker.build` step and install -ext in the Dockerfile directly. It will need some work, though.